dec_unittest: fix vp9 decode unittest failure on BSW/SKL

wudping opened the pull request:
BSW/SKL do not support VP9 decoding, so vp9 decoding leads to core-dump
in decode unittest.
This patch will skip the unittest when detect that the platform dosen't
support VP9.

to fix issue [741](https://github.com/01org/libyami/issues/741)

Signed-off-by: wudping <dongpingx.wu@intel.com>

wudping commented:
please help to review.

xuguangxin commented on decoder/DecoderApi_unittest.cpp:
```diff
@@ -283,6 +291,12 @@ TEST_P(DecodeApiTest, Format_Change)
     while (frames.getFrame(buffer, info)) {
 
         YamiStatus status = decoder->decode(&buffer);
+        if (YAMI_UNSUPPORTED == status) {
```
duplicate code here. 
please use
bool checkStatus(YamiStatus status)
to  remove all duplicates

xuguangxin commented on decoder/DecoderApi_unittest.cpp:
```diff
@@ -283,6 +291,12 @@ TEST_P(DecodeApiTest, Format_Change)
+        if (YAMI_UNSUPPORTED == status) {
+            RecordProperty("skipped", true);
+            std::cout << "[  SKIPPED ] " << getFullTestName()
+                      << " We can't test this in terminal, the stream is unsupported." << std::endl;
```
change this to "hw do not support this decoder"

xuguangxin commented on decoder/DecoderApi_unittest.cpp:
```diff
@@ -297,7 +311,14 @@ TEST_P(DecodeApiTest, Format_Change)
             //send buffer again
-            EXPECT_EQ(YAMI_SUCCESS, decoder->decode(&buffer));
+            status = decoder->decode(&buffer);
+            if (YAMI_UNSUPPORTED == status) {
```
no call decode call need this, we just need check it in first decode call

wudping replied:
hi @xuguangxin ,
This is necessary actually. Because Yami will return YAMI_DECODE_FORMAT_CHANGE when decoding the first frame.

xuguangxin reviewed (changes requested):
Please split core changes and unit test changes

dongpingx-wu committed:
vaapi: return unsupporting flag for unsupporting profile

BSW/SKL do not support VP9 decoding. This patch will return the unsupporting
profile to application side.

Signed-off-by: wudping <dongpingx.wu@intel.com>

dongpingx-wu committed:
unittest_dec: terminate unittest as unsupported profile

Hardware doesn't support the decoder with particular profile.
This patch detects the unsupported profile and then terminate the unittest.

Signed-off-by: wudping <dongpingx.wu@intel.com>
